File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : 'C:/psw/python/input.txt'; 
let input = fs.readFileSync(filePath).toString().trim().split(' ');

const n = parseInt(input[0]);
const m = parseInt(input[1])

let visited = new Array(n).fill(false);
let ptr = [];

function back() {
    if (ptr.length == m) {
        console.log(ptr.join(" "))
    }
    for (let i = 1; i < n + 1; i++) {
        if (visited[i-1] == false){
            visited[i - 1] = true;
            ptr.push(i);
            back();
            visited[i - 1] = false;
            ptr.pop();
        }
    }
}

back()

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
n,m = map(int,input().split())

visited = [0 for _ in range(1,n+1)]
ptr = []

#test
# print(visited)

#back tracking funtion
def back():
    # 백트레킹 종료 조건
    if len(ptr) == m:
        print(' '.join(map(str, ptr)))
        return
    #실제 로직
    for i in range(1,n+1):
        if visited[i-1] == False:
            visited[i-1] = True
            ptr.append(i)
            back()
            visited[i-1] = False
            ptr.pop()

back()